site/_posts/2023-02-10-nextjs-single-sign-on.md

```diff
@@ -0,0 +1,231 @@
+
+## Method 1: Clone the demo repository
+
+If you want to run an already working application, you can clone the demo project from this [GitHub repository](https://github.com/FusionAuth/fusionauth-example-nextjs-single-sign-on) using the following command.
+
+```shell
+$ git clone git@github.com:FusionAuth/fusionauth-example-nextjs-single-sign-on.git
+```
+
+Even though you do not need to actually execute the next steps, we do recommend that you read them to understand what was done in the demo application. After that, continue configuring the [Environment variables](#environment-variables).
+
+## Method 2: Create your own Next.js application
+
+If you want to create your own application instead of using our demo project, you can create a new Next.js application by running the command below.
+
+```shell
+$ npx create-next-app name-of-your-application
+```
+
+If it is your first time creating a Next.js application, you'll be prompted to install the `create-next-app` package. Just type `y` to accept it.
+
+You can answer the questions from the wizard any way you want. Here are the ones that we used in our demo project.
+
+```
+✔ Would you like to use TypeScript with this project? … No
+✔ Would you like to use ESLint with this project? … Yes
+✔ Would you like to use `src/` directory with this project? … Yes
+? Would you like to use experimental `app/` directory with this project? … No
+✔ What import alias would you like configured? … @/*
+```
+
+There are several ways of [implementing authentication with Next.js](https://nextjs.org/docs/authentication, and we'll use [NextAuth.js](https://next-auth.js.org), a complete open-source authentication solution, in this article.
+
+### Installing NextAuth.js
+
+{% include _callout-note.liquid content="We recommend you look at the [NextAuth.js Getting Start guide](https://next-auth.js.org/getting-started/example) for more up-to-date instructions." %}
+
+First, install NextAuth.js via one of the following three methods.
+
+```shell
+# Using npm
+$ npm install next-auth
+
+# Using yarn
+$ yarn add next-auth
+
+# Using pnpm
+$ pnpm add next-auth
+```
+
+Now, you need to create a file named exactly like `[...nextauth].js` in `src/pages/api/auth` (you need to create the `auth` folder there) and configure the [built-in support for FusionAuth](https://next-auth.js.org/providers/fusionauth). This will make every request to `/api/auth/*` handled by NextAuth.js.
+
+```jsx
+import NextAuth from "next-auth"
+import FusionAuthProvider from "next-auth/providers/fusionauth"
+
+export const authOptions = {
+    providers: [
+        FusionAuthProvider({
+            issuer: process.env.FUSIONAUTH_ISSUER,
+            clientId: process.env.FUSIONAUTH_CLIENT_ID,
+            clientSecret: process.env.FUSIONAUTH_CLIENT_SECRET,
+            wellKnown: `${process.env.FUSIONAUTH_URL}/.well-known/openid-configuration`,
```

Comment:
Does this need to have the `clientId` added at the end like you would find in the FusionAuth Application OAuth2 Integration Details?

`http://localhost:9012/.well-known/openid-configuration/{process.env.FUSIONAUTH_TENANT_ID}`